src/schema.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from 'zod'
+
+const walletAddressPattern = /^0x[a-fA-F0-9]{40}$/
```

Comment:
nit: for consistency we should only allow lowercased addresses
```suggestion
const walletAddressPattern = /^0x[a-f0-9]{40}$/
```